src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -437,19 +435,25 @@ bool PosixSignals::chained_handler(int sig, siginfo_t* siginfo, void* context) {
 // handlers, unless invoked with the option "-XX:+AllowUserSignalHandlers".
 //
 
+static void javaSignalHandler(int sig, siginfo_t* info, void* ucVoid);
+static bool javaSignalHandler_inner(int sig, siginfo_t* info, void* ucVoid, bool abort_if_unrecognized);
+
+extern "C" JNIEXPORT int
 #if defined(BSD)
-extern "C" JNIEXPORT int JVM_handle_bsd_signal(int signo, siginfo_t* siginfo,
-                                               void* ucontext,
-                                               int abort_if_unrecognized);
+                         JVM_handle_bsd_signal
```

Comment:
Can we define this using token pasting e.g.

PASTE_TOKENS(JVM_handle, PASTE_TOKENS(INCLUDE_SUFFIX_OS, _signal))

?

Reply:
I find that actually less readable than what we have now. Plus, right now its easier to grep for the function name.